src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs

```diff
@@ -245,6 +246,60 @@ private void ExpandDependency(DocumentBuildParameters parameter, BuildInfo lastB
             }
         }
 
+        private Dictionary<string, FileAttributeItem> ComputeFileAttributes(DocumentBuildParameters parameters)
+        {
+            return (from f in parameters.Files.EnumerateFiles()
+                    let fileKey = ((RelativePath)f.File).GetPathFromWorkingFolder().ToString()
+                    select new FileAttributeItem
+                    {
+                        File = fileKey,
+                        LastModifiedTime = File.GetLastWriteTimeUtc(f.FullPath),
+                        MD5 = File.ReadAllText(f.FullPath).GetMd5String(),
+                    }).ToDictionary(a => a.File);
+        }
+
+        private void LoadChanges(DocumentBuildParameters parameter, DocumentBuildContext context, IReadOnlyDictionary<string, FileAttributeItem> fileAttributes)
+        {
+            var changeItems = context.ChangeDict;
+            if (parameter.Changes != null)
+            {
+                // use user-provided changelist
+                foreach (var pair in parameter.Changes)
+                {
+                    changeItems[pair.Key] = pair.Value;
+                }
+            }
+            else
+            {
+                // get changelist from lastBuildInfo if user doesn't provide changelist
+                string lastAttributesFile = LastBuildInfo.Versions.SingleOrDefault(v => v.VersionName == parameter.VersionName).Attributes;
```

Comment:
actually not because it is already checked in CanIncremental

---
In reply to: [75596649](https://github.com/dotnet/docfx/pull/625#discussion_r75596649) [](ancestors = 75596649)